great_expectations/rule_based_profiler/domain_builder/domain.py

```diff
@@ -1,39 +1,98 @@
+import json
 from typing import Any, Dict, Optional, Union
 
 from great_expectations.core import IDDict
 from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder.inferred_semantic_domain_type import (
+    SemanticDomainTypes,
+)
 from great_expectations.types.base import SerializableDotDict
+from great_expectations.util import filter_properties_dict
 
 
-# TODO: <Alex>ALEX -- multiple inheritance is to be used with caution (potentially fix to use single inheritance)</Alex>
-class Domain(IDDict, SerializableDotDict):
+class DomainKwargs(SerializableDotDict):
+    def to_json_dict(self) -> dict:
+        return convert_to_json_serializable(data=dict(self))
+
+
+class Domain(SerializableDotDict):
     # Adding an explicit constructor to highlight the specific properties that will be used.
-    def __init__(self, domain_kwargs: Optional[Dict[str, Any]] = None):
+    def __init__(
+        self,
+        domain_type: Union[str, MetricDomainTypes, SemanticDomainTypes],
+        domain_kwargs: Optional[Union[Dict[str, Any], DomainKwargs]] = None,
+    ):
+        if not domain_type:
+            raise ValueError("Cannot instantiate Domain (domain_type is required).")
+
+        if domain_kwargs is None:
+            domain_kwargs = DomainKwargs({})
+        elif isinstance(domain_kwargs, dict):
+            domain_kwargs = DomainKwargs(domain_kwargs)
+
+        if isinstance(domain_type, str):
+            if SemanticDomainTypes.has_member_key(key=domain_type):
+                domain_type = SemanticDomainTypes[domain_type]
```

Comment:
I don't think a `domain_type` should include `SemanticDomainTypes`, only `MetricDomainTypes`. We should capture `semantic_domain_type` separately either in `DomainKwargs` or a separate argument to `Domain` (I am thinking the former).

Reply:
Upon discussion - we decided to put SemanticDomainType in a meta dictionary which stores inferred/manual semantic domain type e.g. {"inferred_semantic_domain_type": "string", "manual_semantic_domain_type": "name_of_department", ... any other metadata}

Reply:
@talagluck This has been committed. /cc @anthonyburdi